5.3.Dish_Share.py

```diff
@@ -0,0 +1,25 @@
+
+# define a generator that return the perfect numbers from 1 to infinity
+def perfect_numbers():
+    # define a function that get a number and check if the sum of the divisors of the number is equal to the number
+    # if the sum is equal to the number return True else return False
+    def is_perfect(num):
+        # create a list of the divisors of the number
+        divisors = [i for i in range(1, int(num / 2) + 1) if num % i == 0]
+        # check if the sum of the divisors is equal to the number
+        if sum(divisors) == num:
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.